packages/liveblocks-core/src/room.ts

```diff
@@ -1646,7 +1650,32 @@ export function createRoom<
   }
 
   async function listTextVersions() {
-    return httpClient2.fetch(url`/v2/c/rooms/${config.roomId}/versions`);
+    const response = await httpClient2.fetch(
+      url`/v2/c/rooms/${config.roomId}/versions`
+    );
+
+    if (response.ok) {
+      const json = await (response.json() as Promise<{
+        versions: DateToString<HistoryVersion>[];
+      }>);
+
+      const versions: HistoryVersion[] = json.versions.map(
+        ({ createdAt, ...version }) => {
+          return {
+            createdAt: new Date(createdAt),
+            ...version,
+          };
+        }
+      );
+
+      return {
+        versions,
+      };
+    } else {
+      throw new Error(
+        `There was an error while getting text versions for room ${config.roomId}`
+      );
```

Comment:
@nimeshnayaju I think we can now replace this handling with a "normal" call to `.fetchJson()` here, and just run a "convert to dates" helper function on the success result. It would turn this `Error` into an `HttpError`. Agreed?